src/libraries/Microsoft.PowerFx.Core/Public/Marshal/ObjectMarshalerProvider.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Marshal .Net classes (with fields). This supports strong typing and lazy marshalling. 
+    /// </summary>
+    public class ObjectMarshalerProvider : ITypeMashalerProvider
+    {
+        // $$$ To marshal an object, we need:
+        //  - fxName,
+        //  - .Net type (for getting the marhsaler)
+        //  - getter for runtime value
+        // Customization point 
+        public Func<PropertyInfo, string> PropertyMapperFunc = (propInfo) => propInfo.Name;
```

Comment:
>PropertyMapperFunc [](http://example.com/codeflow?start=42&length=18)

This gives some customization ability. 
If they need radically more control (say allow fields instead of property infos), they can always write their own provider and return a ObjectMarshaler  directly.